qa/tasks/ceph_manager.py

```diff
@@ -110,6 +110,17 @@ def mount_osd_data(ctx, remote, cluster, osd):
             )
 
 
+def log_exc(func):
```

Comment:
just out of curiosity, why do we need to move `log_exc()` to the top level of this file?

Reply:
the previous method misused the self argument as the function. Like so:

```python
In [3]: class C:
     def decorator(self):
         def wrapper(arg):
             return self(arg)
         return wrapper
     @decorator
     def func(self): ...
```

unfortunately that actually works in Python.

Reply:
> the previous method misused the self argument as the function. Like so:

i don't agree. "self" is but a parameter name for the wrapped function from the perspective of `log_exc()`. and `log_exc()` assumes that the wrapped function only accepts a single parameter. it could be anything, but here, it is always "self". that's why i used "self" as the name of the parameter.

i don't see anything wrong or inappropriate here.

> Edit: mypy expects the `self` argument to have the same type as the class like so:

"self" does have the same type of the class.

Reply:
> > the previous method misused the self argument as the function. Like so:
> 
> i don't agree. "self" is but a parameter name for the wrapped function from the perspective of `log_exc()`. and `log_exc()` assumes that the wrapped function only accepts a single parameter. it could be anything, but here, it is always "self". that's why i used "self" as the name of the parameter.

I disagree here. Common sense is that the first argument of class methods are called `self`, because self is handled differently than other parameters. As `log_exc` expected a different type passed in as the first argument, it should live outside of the class. IMO.

> 
> i don't see anything wrong or inappropriate here.
> 
> > Edit: mypy expects the `self` argument to have the same type as the class like so:
> 
> "self" does have the same type of the class.

exactly. That's why it should be moved to toplevel.